splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/actions/DropAliasConstantOperation.java

```diff
@@ -152,6 +152,11 @@ public static void invalidate(AliasDescriptor ad, DependencyManager dm, Language
         }
 
         dm.invalidateFor(ad, invalidationType, lcc);
+        if (ad.getAliasType() == AliasInfo.ALIAS_TYPE_SYNONYM_AS_CHAR) {
+            // invalidate nameTdCache
+            TableKey key = new TableKey(ad.getSchemaUUID(), ad.getName());
+            lcc.getDataDictionary().getDataDictionaryCache().nameTdCacheRemove(key);
```

Comment:
I think this won't work for other RSs, would it?

Can you add a test for that case? Create alias in RS0, use it in RS1, drop it from RS0, see if you can recreate it from RS1

Reply:
@dgomezferro , the fix works for other RSs. This function DropAliasConstantOperation.invalidate() is called by SpliceDatabase.registerDDL()->DDLUtils.preDropAlias(), so all region servers will invoke it.
I've pushed an IT to test it. Thanks!

Reply:
I see, thanks!